tests/common/fixtures/services.py

```diff
@@ -34,7 +37,12 @@ def mock_service(service_class, name):
 
 @pytest.fixture
 def search_index(mock_service):
-    return mock_service(SearchIndexService, "search_index")
+    return mock_service(
+        SearchIndexService,
+        "search_index",
+        spec_set=False,
+        _queue=create_autospec(Queue, spec_set=True, instance=True),
+    )
```

Comment:
Since the Celery task calls `search_index.queue.add_annotations_between_times()` on a mock `SearchIndexService`, the mock `SearchIndexService` needs to have a `_queue`. Normally autospec would not add a `_queue` attr to the mock because [`self._queue` is declared inside the `__init__()`](https://github.com/hypothesis/h/blob/b5f794ba5a0d0f7cb0aaa3c1d93964d273a708b2/h/services/search_index/service.py#L28) so autospec would have to instantiate the class in order to know that `self._queue` exists. There are two workarounds:

1. The fixture could instantiate a real `SearchIndexService` and pass it to `create_autospec()`. When you pass a class to `create_autospec()` with `instance=True` you're asking it to look at a class and return a mock of an object instance of that class. This doesn't work with attrs that're declared inside the `__init__()`. If you instead construct a real `SearchIndexService` instance and pass it to `create_autospec()` (no `instance=True` is needed) you're just asking it to create a mock of this object, and `self._queue` would work

2. What I've gone for here: just pass `_queue=blah` to `create_autospec()` to tell it what to do. This requires `spec_set=False` otherwise it'll complain that "SearchIndexService has no attribute _queue"